autoendpoint/src/extractors/subscription.rs

```diff
@@ -263,8 +268,24 @@ fn validate_vapid_jwt(vapid: &VapidHeaderWithKey, domain: &Url) -> ApiResult<()>
         },
     };
 
-    // Check the signature and make sure the expiration is in the future, but not too far
-    if token_data.claims.exp > (sec_since_epoch() + ONE_DAY_IN_SECONDS) {
+    let exp = if token_data.claims.exp.is_string() {
+        u64::from_str(token_data.claims.exp.as_str().unwrap_or_default()).unwrap_or_default()
+    } else if token_data.claims.exp.is_u64() {
+        token_data.claims.exp.as_u64().unwrap_or_default()
```

Comment:
It looks like `jsonwebtoken` also allows floats, so we probably should too?

https://github.com/Keats/jsonwebtoken/blob/2d2921f/src/validation.rs#L293

Their `fn numeric_type` there set as a `deserialize_with` is a nice way hook into serde to handle multiple input types. I suppose you could copy it over, adding a `visit_str` for this case but I'll admit I'm struggling to figure out why it's using `PhantomData` there (but it's also late Friday..)

Reply:
I dunno, I would prefer that folk follow the spec, which is numeric UTC. Allowing string representation handles bad implementations, but those also stick to UTC. 

This lib is actually more forgiving than we used to be and allows for clock drift

Reply:
The spec calls this NumericDate "A JSON numeric value" and "other than that non-integer values can be represented." so I think that accommodates floats. Blame JSON's number type.